api/schema/deployment_resource.go

```diff
@@ -0,0 +1,67 @@
+package schema
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var MetadataType = types.ObjectType{
+	AttrTypes: map[string]attr.Type{
+		"cloud": types.StringType,
+		"id":    types.StringType,
+	},
+}
+var TagType = types.ObjectType{
+	AttrTypes: map[string]attr.Type{
+		"key":   types.StringType,
+		"value": types.StringType,
+		"cloud": types.StringType,
+	},
+}
+
+var (
+	deploymentResource = Resource{
+		TypeName: "deployment",
+		Schema: resource_schema.Schema{
+			Version:     1,
+			Description: "Groups resources under Cloudsecure deployments",
+			Attributes: map[string]resource_schema.Attribute{
+				IDFieldName: idAttribute,
+				"environment": resource_schema.StringAttribute{
+					MarkdownDescription: "Name of the deployment",
+					Required:            true,
+				},
+				"description": resource_schema.StringAttribute{
+					MarkdownDescription: "Description of Cloudsecure deployment",
+					Optional:            true,
+				},
+				"accounts": resource_schema.ListAttribute{
+					Optional:    true,
+					Description: "cloud accounts that need be a part of this deployment",
+					ElementType: MetadataType,
+				},
+				"regions": resource_schema.ListAttribute{
+					Optional:    true,
+					Description: "regions that need to be associated with this deployment",
+					ElementType: MetadataType,
+				},
+				"vnets": resource_schema.ListAttribute{
+					Optional:    true,
+					Description: "virtual networks that need to be associated with this deployment",
+					ElementType: MetadataType,
+				},
+				"subnets": resource_schema.ListAttribute{
+					Optional:    true,
+					Description: "subnets whose resources need to be associated with this deployment",
+					ElementType: MetadataType,
+				},
+				"tags": resource_schema.ListAttribute{
```

Comment:
Split into `aws_tags` and `azure_tags`, and remove the `cloud` field from the object schema, for consistency with the `aws_tag_to_label` resource's schema, which shouldn't have a `cloud` field either, cf. https://github.com/illumio/terraform-provider-illumio-cloudsecure/pull/116#discussion_r1909438130